kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowLayout.kt

```diff
@@ -57,25 +66,17 @@ class WorkflowLayout(
     newRendering: Any,
     registry: ViewRegistry
   ) {
-    showing?.takeIf { it.canShowRendering(newRendering) }
-        ?.let { it ->
-          it.showRendering(newRendering)
-          return
-        }
-
-    removeAllViews()
-    val newView = registry.buildView(newRendering, this)
+    showing.update(newRendering, registry)
     restoredChildState?.let { restoredState ->
       restoredChildState = null
-      newView.restoreHierarchyState(restoredState)
+      showing.actual!!.restoreHierarchyState(restoredState)
```

Comment:
Could we override `restoreHierarchyState` on `WorkflowViewStub` to automatically delegate to `_actual` if it's not self-referential? Seems like this would be easy to get wrong.

Reply:
It's not necessary in 95% of containers, e.g. note that none of the modal container do. Where it is necessary, like here or in `BackStackContainer`, I don't think there's a general solution. The two use cases are way different.

Reply:
Oh, right, and I didn't even use the stub in BackStackContainer, so I'm down to a single use case. Definitely not ready to generalize, and pretty confident that day to day feature developers don't care.